python/pipeline/reso.py

```diff
@@ -514,10 +511,6 @@ def _make_tuples(self, key):
             self.notify({**key, 'slice': slice_id + 1}, scan) # once per slice
 
     def notify(self, key, scan):
-        # --  notification
-        import matplotlib
-        matplotlib.rcParams['backend'] = 'Agg'
-        import matplotlib.pyplot as plt
```

Comment:
Is `plt` imported somewhere else in the file? Setting it to 'Agg' is ok to do at the top, but I am not sure whether you can get rid of the `import matplotlib.pyplot as plt` import.